holoscribe/src/cli.rs

```diff
@@ -1 +1,142 @@
+use std::num::ParseIntError;
 
+use clap::Parser;
+use regex::Regex;
+use thiserror::Error;
+
+#[derive(Error, Debug, PartialEq)]
+pub enum CliError {
+    #[error("Invalid int specification for size")]
+    InvalidSizeInt(ParseIntError),
+    #[error("Invalid size specification, please use the format: width[xheight](mm|cm|m)")]
+    InvalidSize,
+}
+
+#[derive(Parser)]
+#[command(author, version, about, long_about = None)]
+#[command(propagate_version = true)]
+/// A command line utility that take an .obj model and produce a .svg representations of lines to etch
+/// to create a holographic image of that model
+pub struct Args {
+    /// Input file. Expects a .obj file
+    #[arg(short, long)]
+    pub input: String,
+
+    /// Output file. Expects a .svg file
+    #[arg(short, long)]
+    pub output: String,
+
+    /// The size of the image you which to create. Requires the format: width[xheight](mm|cm|m)
+    #[arg(short, long, value_parser=parse_size)]
+    pub size: Size,
+
+    /// The density of lines etched per mm. Defaults to 1.
+    #[arg(long, default_value_t = 1)]
+    pub stroke_density: usize,
+}
+
+/// Represents a size in millimeters
+#[derive(Debug, Clone, PartialEq)]
+pub struct Size {
```

Comment:
If this struct is going to be used by other parts of this project, we might want to consider renaming this to `canvas_size` to avoid any ambiguity.